Promote PodTopologySpread to stable in 1.19 (#21115)
kubernetes/website: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md

> ## Prerequisites

Reviewer: @Huang-Wei ,
Are there prerequisites?

Reviewer: Could you rename this heading?

You can define one or multiple `topologySpreadConstraint` to instruct the kube-scheduler how to place each incoming Pod in relation to the existing Pods across your cluster. The fields are:

- **maxSkew** describes the degree to which Pods may be unevenly distributed.
  It's the maximum permitted difference between the number of matching Pods in
  any two topology domains of a given topology type. It must be greater than
  zero. Its semantics differs according to the value of `whenUnsatisfiable`:
  - when `whenUnsatisfiable` equals to "DoNotSchedule", `maxSkew` is the maximum
    permitted difference between the number of matching pods in the target
    topology and the global minimum.

>   - when `whenUnsatisfiable` equals to "ScheduleAnyway", scheduler gives higher

Reviewer 2: _nit (OK to skip / fix after merge):_
“is equal to” or “equals”, not “equals to” (that last phrasing is not idiomatic English)

- For `PodAntiAffinity`, only one Pod can be scheduled into a
  single topology domain.

> The "PodTopologySpread" feature provides flexible options to distribute Pods evenly across different

Reviewer 2: You don't need to name feature gates for graduated features.

topology domains - to achieve high availability or cost-saving. This can also help on rolling update
workloads and scaling out replicas smoothly. See [Motivation](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/895-pod-topology-spread#motivation) for more details.

## Known Limitations

- Scaling down a Deployment may result in imbalanced Pods distribution.

> - Pods matched on tainted nodes are respected. See [Issue 80921](https://github.com/kubernetes/kubernetes/issues/80921)

Reviewer 2: Is this line still relevant?

Author: Yes.

- For `PodAntiAffinity`, only one Pod can be scheduled into a
  single topology domain.

> To fill the gap, you can specify topology spread constraints to distribute Pods
> across different topology domains flexibly - to achieve either high availability

Reviewer 3: ```suggestion
For finer control, you can specify topology spread constraints to distribute Pods
across different topology domains - to achieve either high availability
```
